src/impl/MilvusClientImpl.cpp

```diff
@@ -135,7 +135,27 @@ MilvusClientImpl::GetCollectionStatistics(const std::string& collection_name, co
 
 Status
 MilvusClientImpl::ShowCollections(const std::vector<std::string>& collection_names, CollectionsInfo& collections_info) {
-    return Status::OK();
+    if (connection_ == nullptr) {
+        return Status(StatusCode::NOT_CONNECTED, "Connection is not ready!");
+    }
+    proto::milvus::ShowCollectionsRequest rpc_request;
+    for (auto& collection_name : collection_names) {
+        rpc_request.add_collection_names(collection_name);
+    }
+
+    proto::milvus::ShowCollectionsResponse response;
+    Status ret = connection_->ShowCollections(rpc_request, response);
+    if(ret.IsOk())
+    {
+        for(size_t i = 0; i < response.collection_ids_size(); i++)
+        {
+            collections_info.push_back(CollectionInfo(*response.collection_names[i], 
```

Comment:
Compile error here. You need to call  `response.collection_names(i) `

Reply:
The ShowCollections() has two options:
- If the collection_names is empty, list all collections. Pass ShowType=All to the ShowCollectionsRequest.
- If the collection_names is specified, additionally return the specified collection's loading process state. Pass ShowType=InMemory to the ShowCollectionsRequest.

Reply:
Yes i figured this one out. Ill puah the changes.

Reply:
Prefer using emplace_back for reducing memory copy

Reply:
Okay